File: ScriptGraph/TemplateScriptNode/Sqrt/CGSqrt.lua
local CGSqrt= CGSqrt or {}
CGSqrt.__index = CGSqrt

function CGSqrt.new()
    local self = setmetatable({}, CGSqrt)
    self.inputs = {}
    return self
end

function CGSqrt:setInput(index, func)
    self.inputs[index] = func
end

function CGSqrt:getOutput(index)
    return math.sqrt(self.inputs[0]())
end

return CGSqrt

File: ScriptGraph/TemplateScriptNode/Sqrt/CGSqrt.js
/**
 * @file CGSqrt.js
 * @author liujiacheng
 * @date 2021/8/23
 * @brief CGSqrt.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const {BaseNode} = require('./BaseNode');
const Amaz = effect.Amaz;

class CGSqrt extends BaseNode {
  constructor() {
    super();
  }

  setNext(index, func) {
    this.nexts[index] = func;
  }

  setInput(index, func) {
    this.inputs[index] = func;
  }

  getOutput() {
    return Math.sqrt(this.inputs[0]());
  }
}

exports.CGSqrt = CGSqrt;
